### PR TITLE
metrics(ticdc): fix slowest table metrics in scheduler

### LINKED ISSUE #10482: the ticdc_scheduler_slow_table_puller_resolved_ts_lag metric is incorrect
### What did you see instead?

ref https://github.com/pingcap/tiflow/issues/10462#issuecomment-1893215086

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #10482

### What is changed and how it works?
1. [Record the correct progress in slowestTablePullerResolvedTs](https://github.com/pingcap/tiflow/pull/10484/files#diff-ec8c2629076697b5b91801313d52d043e53e4418f0317c4b944d380cdaca1466R861-R864)
2. [For tables with equal checkpoints, set the table with the smallest resovedTs to the slowest table](https://github.com/pingcap/tiflow/pull/10484/files#diff-ec8c2629076697b5b91801313d52d043e53e4418f0317c4b944d380cdaca1466R630-R631)

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by hongyunyan — APPROVED** [review]
LGTM

**Review by asddongmen — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

File: cdc/scheduler/internal/v3/replication/replication_manager.go [hunk 1]
```diff
@@ -625,9 +626,12 @@ func (r *Manager) AdvanceCheckpoint(
 				}
 
 				// Find the minimum checkpoint ts and resolved ts.
-				if watermark.CheckpointTs > table.Checkpoint.CheckpointTs {
+				if watermark.CheckpointTs > table.Checkpoint.CheckpointTs ||
+					(watermark.CheckpointTs == table.Checkpoint.CheckpointTs &&
```
Comment (asddongmen) [inline cdc/scheduler/internal/v3/replication/replication_manager.go:630]: Please add some comment why we need to check `watermark.CheckpointTs == table.Checkpoint.CheckpointTs`?

**Pull request merged**